apis/v1beta1/vspheredeploymentzone_types.go

```diff
@@ -28,6 +28,95 @@ const (
 	DeploymentZoneFinalizer = "vspheredeploymentzone.infrastructure.cluster.x-k8s.io"
 )
 
+// VSphereDeploymentZone's Ready condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// VSphereDeploymentZoneReadyV1Beta2Condition is true if the VSphereDeploymentZone's deletionTimestamp is not set, VSphereDeploymentZone's
+	// VCenterAvailable, PlacementConstraintReady and FailureDomainValidated conditions are true.
+	VSphereDeploymentZoneReadyV1Beta2Condition = clusterv1.ReadyV1Beta2Condition
+
+	// VSphereDeploymentZoneReadyV1Beta2Reason surfaces when the VSphereDeploymentZone readiness criteria is met.
+	VSphereDeploymentZoneReadyV1Beta2Reason = clusterv1.ReadyV1Beta2Reason
+
+	// VSphereDeploymentZoneNotReadyV1Beta2Reason surfaces when the VSphereDeploymentZone readiness criteria is not met.
+	VSphereDeploymentZoneNotReadyV1Beta2Reason = clusterv1.NotReadyV1Beta2Reason
+
+	// VSphereDeploymentZoneReadyUnknownV1Beta2Reason surfaces when at least one VSphereDeploymentZone readiness criteria is unknown
+	// and no VSphereDeploymentZone readiness criteria is not met.
+	VSphereDeploymentZoneReadyUnknownV1Beta2Reason = clusterv1.ReadyUnknownV1Beta2Reason
+)
+
+// VSphereDeploymentZone's VCenterAvailable condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// VSphereDeploymentZoneVCenterAvailableV1Beta2Condition documents the status of vCenter for a VSphereDeploymentZone.
+	VSphereDeploymentZoneVCenterAvailableV1Beta2Condition = "VCenterAvailable"
+
+	// VSphereDeploymentZoneVCenterAvailableV1Beta2Reason surfaces when the vCenter for a VSphereDeploymentZone is available.
+	VSphereDeploymentZoneVCenterAvailableV1Beta2Reason = clusterv1.AvailableV1Beta2Reason
+
+	// VSphereDeploymentZoneVCenterUnreachableV1Beta2Reason surfaces when the vCenter for a VSphereDeploymentZone is unreachable.
+	VSphereDeploymentZoneVCenterUnreachableV1Beta2Reason = "VCenterUnreachable"
+
+	// VSphereDeploymentZoneVCenterAvailableDeletingV1Beta2Reason surfaces when the vCenter for a VSphereDeploymentZone is being deleted.
+	VSphereDeploymentZoneVCenterAvailableDeletingV1Beta2Reason = clusterv1.DeletingV1Beta2Reason
+)
```

Comment:
in https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/3406 I have `Available` and `NotAvailable` as reasons for the `VCenterAvalable` condition on VSphereVM, wondering if we should align

Reply:
Wondering if we should merge all 3 PRs and then do one PR to align (instead of syncing up all 3 open PRs)

Reply:
Discussed, we're fine with the additional Deleting reason here